clang/lib/Basic/Targets/OSTargets.h

```diff
@@ -352,6 +352,8 @@ class LLVM_LIBRARY_VISIBILITY LinuxTargetInfo : public OSTargetInfo<Target> {
     case llvm::Triple::x86:
     case llvm::Triple::x86_64:
       this->HasFloat128 = true;
+      if (!Triple.isAndroid())
+        this->DecimalFloatEnablementAndMode = DecimalFloatMode::BID;
```

Comment:
Here we are deciding that if Linux and not Android, then encoding is BID. When could it be DPD?

Reply:
This is temporary-ish; it enables DFP support by default on Linux systems that use libgcc. Android is excluded for now since it uses compiler-rt and that RT lacks DFP support at present. As far as I know, DPD is only really used on IBM architectures and OSs like AIX on Power and z/OS.

Reply:
Does it make sense to put a comment here explaining the exception even if it is "temporary"?

Reply:
Yes, it does, good suggestion. I'll add that and then land this change.